pkg/apis/flowcontrol/validation/validation.go

```diff
@@ -73,25 +75,41 @@ var supportedLimitResponseType = sets.NewString(
 // ValidateFlowSchema validates the content of flow-schema
 func ValidateFlowSchema(fs *flowcontrol.FlowSchema) field.ErrorList {
 	allErrs := apivalidation.ValidateObjectMeta(&fs.ObjectMeta, false, ValidateFlowSchemaName, field.NewPath("metadata"))
-	allErrs = append(allErrs, ValidateFlowSchemaSpec(&fs.Spec, field.NewPath("spec"))...)
+	specPath := field.NewPath("spec")
+	allErrs = append(allErrs, ValidateFlowSchemaSpec(fs.Name, &fs.Spec, specPath)...)
+	if mand, ok := internalbootstrap.MandatoryFlowSchemas[fs.Name]; ok {
+		// Check for almost exact equality.  This is a pretty
+		// strict test, and it is OK in this context because both
+		// sides of this comparison are intended to ultimately
+		// come from the same code.
+		if !apiequality.Semantic.DeepEqual(fs.Spec, mand.Spec) {
```

Comment:
i thought we agreed on not to forbid user behaviors from validation, instead we periodically reconcile by bootstrap controller from the post-start-hook.

Reply:
It's not an either-or.  The controller makes sure that good stuff happens, and we also use validation to prevent the admins from requesting bad stuff.  We discussed this in the recent Friday meeting with just you and I, and I thought you agreed.

Reply:
Both pieces are needed, this validation could go here or in an admission plugin. The latter might be better but not so much that I'd ask for this to change.

Reply:
I thought we agreed that a controller that periodically makes sure the right mandatory objects exist would be sufficient, so why is the validation or admission plugin _needed_?  I see only a UX improvement: making it difficult (but not impossible in mixed-version scenarios) for admins to create a state that will soon be undone.